SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -335,6 +410,22 @@ USpatialLatencyTracer::TraceSpan* USpatialLatencyTracer::GetActiveTrace()
 	return TraceMap.Find(ActiveTraceKey);
 }
 
+USpatialLatencyTracer::TraceSpan* USpatialLatencyTracer::GetActiveTraceOrReadPayload(const FSpatialLatencyPayload& Payload)
+{
+	USpatialLatencyTracer::TraceSpan* ActiveTrace = GetActiveTrace();
+	if (ActiveTrace == nullptr)
+	{
+		// Try read the trace from the payload
+		TraceKey Key = ReadTraceFromSpatialPayload(Payload);
```

Comment:
What if the Key is InvalidTraceKey?

Reply:
That's ok - i'll add a warn log about it though because it'll mean that the payload wasn't passed (which is required for native tracing)